zulipterminal/model.py

```diff
@@ -730,6 +730,15 @@ def toggle_stream_pinned_status(self, stream_id: int) -> bool:
     def is_user_subscribed_to_stream(self, stream_id: int) -> bool:
         return stream_id in self.stream_dict
 
+    def is_user_view_in_removed_stream(self, stream_id: int) -> bool:
+        for stream in self.pinned_streams:
+            if stream['id'] == stream_id:
+                return False
+        for stream in self.unpinned_streams:
+            if stream['id'] == stream_id:
+                return False
+        return True
```

Comment:
Note that this should be equivalent to the (simpler) method just above it, as long as the stream_dict is updated appropriately - which we really should do (in addition to the muted stream tracking).

Currently stream_dict is just subscribed streams, though is the original data from the server as opposed to the smaller/simpler data used by the UI (StreamData).

Note that if we did use this method, the name is much more specific than is warranted by the logic - I'd want a simpler cleaner name :)

Reply:
The stream_dict can't be updated in case of removal of streams. This causes any other action related to that stream to fail, like editing, or adding a message. I added the UI warnings to counter this in the first place. I have updated the commits and the UI warnings are now part of a separate commit.